high_card.rb

```diff
@@ -1,5 +1,45 @@
-# Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
-suits = [ "hearts", "spades", "clubs", "diamonds" ]
+require "pry"
 
-players = []
+def build_deck
+@values = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
+@suits = [ "hearts", "spades", "clubs", "diamonds" ]
+@deck = []
+
+@values.each_with_index do |v, i| #each_with_index, adds index value to each item in array
+  @suits.each do |s|
+    @deck.push({score: i, value: v, suit: s})
+  end
+end
+return @deck.shuffle!
+end
+
+deck = build_deck #call the build_deck method
+
+@players = []
+def get_players players
```

Comment:
No need to pass players in.  Just create them and return them.
`players = get_players`  would do what you need for creating them.  Then you just pass that var when you need it.  Like you did on line 37.